### PR TITLE
PT-2192: Tests for patient indexing

### PR DESCRIPTION
**Pull request by buske**
PR originally by @veronikaslc, patched by @buske

### CODE REVIEW

File: components/patient-data/indexing/pom.xml [hunk 1]
```diff
@@ -136,5 +142,10 @@
       <version>${xwiki.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>javax.servlet</groupId>
+      <artifactId>javax.servlet-api</artifactId>
+      <version>3.0.1</version>
```
Comment (sdumitriu) [inline components/patient-data/indexing/pom.xml]: Don't specify the version, this is a managed dependency.

Also, it's only needed transitively during tests, so add `<scope>test</scope>`.

File: components/patient-data/indexing/src/test/java/org/phenotips/data/indexing/internal/PatientEventListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+public class PatientEventListenerTest {
```
Comment (sdumitriu) [inline components/patient-data/indexing/src/test/java/org/phenotips/data/indexing/internal/PatientEventListenerTest.java]: Wrong codestyle.

**Comment by buske** [comment]
Thanks, @sdumitriu. I've committed fixes for those two comments.

File: components/patient-data/indexing/pom.xml [hunk 3]
```diff
@@ -27,6 +27,12 @@
   </parent>
   <artifactId>patient-data-indexing</artifactId>
   <name>PhenoTips - Patient data - Solr indexing</name>
+
+  <properties>
+    <!-- Instruction Ratio of tests is actually 100%. Maven is Bugged -->
```
Comment (sdumitriu) [inline components/patient-data/indexing/pom.xml]: No, there really was a missing branch, see acb2489033c3f6cd715f1cde18bcfe662b05730e.

File: components/patient-data/indexing/src/test/java/org/phenotips/data/indexing/internal/PatientEventListenerTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    public MockitoComponentMockingRule<EventListener> mocker =
+        new MockitoComponentMockingRule<EventListener>(PatientEventListener.class);
+
+    @Mock
+    private PatientIndexer patientIndexer;
+
+    @Mock
+    private Patient patient;
+
+    private PatientEventListener patientEventListener;
```
Comment (sdumitriu) [inline components/patient-data/indexing/src/test/java/org/phenotips/data/indexing/internal/PatientEventListenerTest.java]: It's better to use the generic interface which defines the "contract" when testing.

File: components/patient-data/indexing/src/test/java/org/phenotips/data/indexing/internal/PatientEventListenerTest.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+        this.patientEventListener = (PatientEventListener) this.mocker.getComponentUnderTest();
+        this.patientIndexer = this.mocker.getInstance(PatientIndexer.class);
+    }
+
+    @Test
+    public void deletePatientTest()
+    {
+        Event patientDeleteEvent = mock(PatientDeletedEvent.class);
+        doReturn(this.patient).when((PatientEvent) patientDeleteEvent).getPatient();
```
Comment (sdumitriu) [inline components/patient-data/indexing/src/test/java/org/phenotips/data/indexing/internal/PatientEventListenerTest.java]: I find the `when(something).thenSomething()` mocking style easier to comprehend.

**Pull request merged**